config/routes.rb

```diff
@@ -10,6 +10,7 @@ def put(*_args); end
   scope :as => :api, :module => "api", :path => prefix do
     match "/v0/*path", :via => [:delete, :get, :options, :patch, :post], :to => redirect(:path => "/#{prefix}/v0.1/%{path}", :only_path => true)
     namespace :v0x1, :path => "v0.1" do
+      get "/openapi.json", :to => "root#openapi"
```

Comment:
Since this lives in `public` you shouldn't need to add the controller.  
```
get "/openapi.json",  :to => redirect("/catalog/v0.1.0/openapi.json")
```

Reply:
The only downer is the redirect, but it does remove the need to add the controller.

Reply:
@syncrou Yes we tried that and a 321 doesn't work well with API's.

Reply:
Does this match what the topo is doing?